### DISCUSSION TITLE
React App on Github Pages is Blank and Returns Error 404

### DISCUSSION THREAD

**Discussion by toyosicodes** [community/community#22392] [Other Feature Feedback, Questions, & Ideas]
<p>Hi everyone,</p>
<p>I deployed a react app to Github Pages and apparently it was successful, but the site shows only a blank white page. The console says ‘Failed to load resource: the server responded with a status of 404 ()’. Here is my repo: <a href="https://github.com/toyosicodes/memory-game.git" class="inline-onebox" rel="noopener nofollow ugc">GitHub - toyosicodes/memory-game</a>. The link to the site itself is <a href="https://github.com/toyosicodes/memory-game.git" class="inline-onebox" rel="noopener nofollow ugc">GitHub - toyosicodes/memory-game</a>.</p>
<p>I’d appreciate any help. I can’t figure out what I’m doing wrong. I’ve spent hours reading topics about this issue and applied the recommended solutions, but they haven’t worked in my case.</p>

**Comment by Katsute** [comment, marked as answer]
<p>GitHub pages only supports <em>static sites</em>, it does not run <code>npm start</code></p>

**Comment by toyosicodes** [comment]
<p>I’m sorry, I don’t understand this</p>

**Comment by Katsute** [comment]
<p>GitHub pages only serves files that are committed to the repository, any files that are not committed (ex: node_modules, dist) will not be available on the site and will cause a 404 not found.</p>

**Comment by toyosicodes** [comment]
<p>Thanks for the explanation. So how can I fix the problem? Because there has to be a way to commit node_modules, right?</p>

**Comment by Katsute** [comment]
<p>Upon closer inspection it seems that the issue is caused by the links to your sources. The node_modules isn’t the problem.</p>
<p><strong>Take this link for example:</strong></p>
<p><strong><code>/toyosicodes/memory-game.github.io/favicon.ico</code></strong></p>
<p>When you deploy the site to GitHub pages it serves it at <code>https://toyosicodes.github.io/memory-game/</code>, the index.html file is expecting to find the favicon at:</p>
<p><code>https://toyosicodes.github.io</code><strong><code>/toyosicodes/memory-game.github.io/favicon.ico</code></strong></p>
<p><strong>The correct link would be:</strong></p>
<p><code>favicon.ico</code> or <code>/memory-game/favicon.ico</code></p>
<p>Any of these changes would correct the url to:</p>
<p><code>https://toyosicodes.github.io/memory-game/</code><strong><code>favicon.ico</code></strong><br>
<code>https://toyosicodes.github.io</code><strong><code>/memory-game/favicon.ico</code></strong></p>

**Comment by toyosicodes** [comment]
<p>I really appreciate you helping me. I have to confess, though, that I have no idea how to make the changes you’re suggesting. If you could guide me on that I’d be really grateful. I forgot to add that I’m a beginner when I originally wrote the topic :sweat_smile:</p>

**Comment by Katsute** [comment]
<p>The <code>%PUBLIC_URL%</code> inside the href/src is incorrect, take this line for example:</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/toyosicodes/memory-game/blob/c8b97d93ee3da3507de487be88ad980290a2fac1/public/index.html#L5">
  <header class="source">

      <a href="https://github.com/toyosicodes/memory-game/blob/c8b97d93ee3da3507de487be88ad980290a2fac1/public/index.html#L5" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/toyosicodes/memory-game/blob/c8b97d93ee3da3507de487be88ad980290a2fac1/public/index.html#L5" target="_blank" rel="noopener nofollow ugc">toyosicodes/memory-game/blob/c8b97d93ee3da3507de487be88ad980290a2fac1/public/index.html#L5</a></h4>



  <pre class="onebox">    <code class="lang-html">
      <ol class="start lines" start="1" style="counter-reset: li-counter 0 ;">
          <li>&lt;!DOCTYPE html&gt;</li>
          <li>&lt;html lang="en"&gt;</li>
          <li>  &lt;head&gt;</li>
          <li>    &lt;meta charset="utf-8" /&gt;</li>
          <li class="selected">    &lt;link rel="icon" href="%PUBLIC_URL%/favicon.ico" /&gt;</li>
          <li>    &lt;meta name="viewport" content="width=device-width, initial-scale=1" /&gt;</li>
          <li>    &lt;meta name="theme-color" content="#000000" /&gt;</li>
          <li>    &lt;meta</li>
          <li>      name="description"</li>
          <li>      content="Web site created using create-react-app"</li>
          <li>    /&gt;</li>
          <li>    &lt;link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" /&gt;</li>
          <li>    &lt;!--</li>
          <li>      manifest.json provides metadata used when your web app is installed on a</li>
          <li>      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>After building, the link changes to <strong><code>/toyosicodes/memory-game.github.io</code></strong><code>/favicon.ico</code> where the site then expects to find a file at <code>https://toyosicodes.github.io</code><strong><code>/toyosicodes/memory-game.github.io/favicon.ico</code></strong>, which is results in a 404 because their is no file at the link.</p>
<p>The <code>%PUBLIC_URL%</code> should instead be <strong><code>/memory-game</code></strong>, where upon building it would change to <strong><code>/memory-game</code></strong><code>/favicon.ico</code> then the site would expect to find that icon at <code>https://toyosicodes.github.io</code><strong><code>/memory-game</code></strong><code>/favicon.ico</code></p>

**Comment by Hariprakash-va** [comment]
Hi,

Im also facing same error my output pages blank
This is my repo:  https://github.com/Hariprakash-va/Todolist/tree/gh-pages
Could you pls help me?

**Comment by vramdal** [comment]
For anyone landing here with this problem in their Create React App: 

You need to add the `homepage` property to your `package.json`. The value should be the URL of your site, e.g. `https://yourname.github.io/your-repo`

If you don't do this, the references to your static files (`bundle.js`, `*.css`, etc) will be wrong.

Documentation: https://create-react-app.dev/docs/deployment/#step-1-add-homepage-to-packagejson

**Reply by me-ibad** [reply]
Hello I follow all these steps with home page as well can you please check my repo and see if any other problems I am try to fix this issue for the last 2 days
https://github.com/me-ibad/bookingCalender

**Comment by NikhilSethuram** [comment]
Hi im facing the same issue. This is my repo
https://github.com/NikhilSethuram/nstPortfolio

**Reply by me-ibad** [reply]
How did you fix? iam  still facing this issue here is may simple react app
https://github.com/me-ibad/bookingCalender

**Comment by jfullstackdev** [comment]
for those who are wondering, you can use React to create a _static_ site only but you need to deploy it first and make sure the build output is in the correct place, then GitHub Pages will start hosting it

actually if you know much of advanced, you can deploy a full stack application but the backend service will be in a different hosting site, this is also true if you only want to develop a UI for, say, a GitHub App using its own API, the frontend will be done in React and be hosted in Pages

these are the steps from Copilot,

![image](https://github.com/community/community/assets/47092464/b7e83b38-2028-46ee-8187-7d2e8b21b959)

**Reply by me-ibad** [reply]
Hi I am using same step but got 404 assets error here is my repo
https://github.com/me-ibad/bookingCalender

**Reply by jfullstackdev** [reply]
you've missed the actual repo name `bookingCalendar` to be part of the URL, as you are not in the root, so the correct URL is 

https://me-ibad.github.io/bookingCalendar/src/main.tsx

as for `vite.svg`, there is no such file in the root in the main branch, it's in `gh-pages` and it's not the one being hosted for sure because I tried the main.tsx, and I can find it on the page, so it means it's hosting in the default branch, attempting to access it through default branch is a sure 404

**Reply by aidanmonty99** [reply]
Hi,

I also followed the above steps, but when I go to my web page I get a blank screen and a 404 error on the following files:
![404_errors](https://github.com/user-attachments/assets/8855a19b-7d69-46f5-a4ae-d914971218a1)

My repo is here: https://github.com/aidanmonty99/aidanmontydotcom

Thanks

**Reply by aidanmonty99** [reply]
Nevermind, I got it to work by removing the repo name from the end of my homepage URL, leaving it as "https://aidanmonty99.github.io/". I assume this is because the app is already hosted in the root of the repo, but I'm not completely sure.

**Comment by Gabrielttbr** [comment]
**Good afternoon, everyone.**

To deploy a React application with multiple routes on GitHub Pages, you first need to configure your routes by adding the baseRoute to them.

Example:

```
<ContextShoppingCart.Provider value={{
    shoppingCart,
    setShoppingCart
  }}>
  <BrowserRouter basename="your-repository-github">
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/shopping-cart" element={<ShoppingCart />} />
      <Route path="*" element={<NotFound />} />
    </Routes>
  </BrowserRouter>
</ContextShoppingCart.Provider>
```

Next, you need to add a configuration in your vite.config.js:

JavaScript
```

export default {
  base: '/your-repository-github/',
  plugins: [react()]
}
```
Then, you need to follow the installation steps from https://github.com/gitname/react-gh-pages.

Goodbye!

**Reply by FDiskas** [reply]
In index.html you can actually write
```html
<base href="%BASE_URL%" />
```
it will be automatically replaced with the value in vite.config.js `base` key

**Comment by federicoisa** [comment]
buenas, como estan?, intente ya muchas veces y tambien guiandome con videos, con chatgpt y hasta de un curso y no pude, cuando despliego en github pages mi app, en el localhost funciona excelente, le hice cambios a las rutas de imagenes, actualice el docs varias veces luego de cada cambio, las rutas de las paginas tambien las actualice, el tema es que creo que lo estoy haciendo mal porque por mas que cambie las rutas, parece que para github pages ignora y se comporta mal y yo no estoy encontrando la manera de hacer que las rutas sean correctas, les dejo mi repositorio para que lo vean y el link de mi portfolio, les agradeceria su ayuda, lo necesito y de paso me gustaria aprender que error estoy cometiendo.
https://federicoisa.github.io/Portfolio-FedericoIsa/
git@github.com:federicoisa/Portfolio-FedericoIsa.git
